optuna/study/study.py

```diff
@@ -263,6 +264,31 @@ def objective(trial):
         Returns:
             A list of :class:`~optuna.trial.FrozenTrial` objects.
         """
+        return self._get_trials(deepcopy, states, use_cache=False)
+
+    def _get_trials(
+        self,
+        deepcopy: bool = True,
+        states: Optional[Container[TrialState]] = None,
+        use_cache: bool = False,
+    ) -> List[FrozenTrial]:
+        if use_cache:
+            if self._thread_local.cached_all_trials is None:
+                self._thread_local.cached_all_trials = self._fetch_trials(deepcopy=False)
+            trials = self._thread_local.cached_all_trials
+            if states is not None:
+                filtered_trials = [t for t in trials if t.state in states]
+            else:
+                filtered_trials = trials
+            return copy.deepcopy(filtered_trials) if deepcopy else filtered_trials
+
+        return self._fetch_trials(deepcopy=deepcopy, states=states)
+
+    def _fetch_trials(
+        self,
+        deepcopy: bool = True,
+        states: Optional[Container[TrialState]] = None,
+    ) -> List[FrozenTrial]:
```

Comment:
Although this is a private method, it's a bit confusing because `_get_trials` and `_fetch_trials` looks quite similar in terms of the interface and the method names. How about removing this method and inlining into `_get_trials`? It doesn't increase the code size.

Reply:
makes sense. updated the code.